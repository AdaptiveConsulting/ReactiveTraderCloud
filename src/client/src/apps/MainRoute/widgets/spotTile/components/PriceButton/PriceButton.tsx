import React from 'react'
import { Direction } from 'rt-types'
import {
  TradeButton,
  Price,
  BigWrapper,
  DirectionLabel,
  Big,
  Pip,
  Tenth,
  ExpiredPrice,
} from './styled'

interface PriceButtonProps {
  big?: number
  pip?: number
  tenth?: number
  rawRate?: number
  direction?: Direction
  handleClick?: () => void
  priceAnnounced?: boolean
  disabled?: boolean
  expired?: boolean
  currencyPairSymbol: string
}

const renderPips = (pips: number) => (pips.toString().length === 1 ? `0${pips}` : pips)
const getBigFigureDisplay = (bigFigure: number, rawRate: number) =>
  bigFigure === Math.floor(rawRate) ? `${bigFigure}.` : bigFigure.toString()
const renderBigFigureDisplay = (bigFigureDisplay: string) =>
  bigFigureDisplay.toString().length === 3 ? `${bigFigureDisplay}0` : bigFigureDisplay

const PriceButtonComp: React.FC<PriceButtonProps> = ({
  big = 0,
  pip = 0,
  tenth = 0,
  rawRate = 0,
  direction = Direction.Buy,
  handleClick = () => {},
  priceAnnounced,
  disabled = false,
  expired = false,
  currencyPairSymbol,
}) => {
  const bigFigure = getBigFigureDisplay(big, rawRate)
  const hasPrice = rawRate !== 0
  const isDisabled = disabled || !hasPrice
  return (
    <TradeButton
      direction={direction}
      onClick={handleClick}
      priceAnnounced={priceAnnounced}
      disabled={isDisabled}
      data-qa="price-button__trade-button"
      data-qa-id={`direction-${direction.toLowerCase()}-${currencyPairSymbol.toLowerCase()}`}
    >
      <Price disabled={isDisabled}>
        <BigWrapper>
          <DirectionLabel>{direction.toUpperCase()}</DirectionLabel>
          <Big data-qa="price-button__big">
            {hasPrice ? renderBigFigureDisplay(bigFigure) : '-'}
          </Big>
        </BigWrapper>
        {hasPrice && (
          <React.Fragment>
            <Pip data-qa="price-button__pip">{renderPips(pip)}</Pip>
            <Tenth data-qa="price-button__tenth">{tenth}</Tenth>
          </React.Fragment>
        )}
      </Price>
      {expired && <ExpiredPrice data-qa="price-button__expired">Expired</ExpiredPrice>}
    </TradeButton>
  )
}

export default PriceButtonComp
