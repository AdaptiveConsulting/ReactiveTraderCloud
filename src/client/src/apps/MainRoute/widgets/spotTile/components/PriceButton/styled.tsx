import styled, { keyframes, css } from 'styled-components/macro'
import { Theme } from 'rt-theme'
import { Direction } from 'rt-types'

const backgroundEffectKeyframes = ({
  direction,
  theme,
}: {
  direction: Direction
  theme: Theme
}) => keyframes`
  5% {
    background-color: ${theme.colors.spectrum.uniqueCollections[direction].base};
    color: white;
  }
  80% {
    background-color: ${theme.colors.spectrum.uniqueCollections[direction].base};
    color: white;
  }
`

const getAnimationCSSProperty = (props: { direction: Direction; theme: Theme }) => css`
  animation: ${backgroundEffectKeyframes(props)} 5s;
`

const backgroundEffect = ({ priceAnnounced, ...rest }: { priceAnnounced: boolean }) =>
  priceAnnounced ? getAnimationCSSProperty(rest as { direction: Direction; theme: Theme }) : ''

export const TradeButton = styled.button<{
  direction: Direction
  priceAnnounced: boolean
  isAnalyticsView: boolean
  expired?: boolean
}>`
  background-color: ${({ theme }) => theme.core.lightBackground};
  border-radius: 3px;
  color: ${({ theme, priceAnnounced, direction }) =>
    priceAnnounced ? theme.colors.spectrum.uniqueCollections[direction].base : 'inherit'};
  transition: background-color 0.2s ease;
  cursor: pointer;
  border: none;
  outline: none;
  height: 59px;
  min-width: 125px;
  padding: ${({ expired }) =>
    expired ? '0.6rem 1.5rem 0.4rem 1.5rem' : '0.55rem 1.5rem 0.6rem 1.5rem'};
  margin-bottom: 2px;
  ${backgroundEffect}

  ${({ theme, direction, disabled, priceAnnounced }) =>
    disabled && !priceAnnounced
      ? `
  cursor: initial;
  pointer-events: none;
    `
      : `

  &:hover {
    background-color: ${theme.colors.spectrum.uniqueCollections[direction].base};
    color: ${theme.white};
  }
  `};
`
const Box = styled.div`
  padding: 0;
  margin: 0;
`

export const DirectionLabel = styled(Box)<{ priceAnnounced?: boolean }>`
  opacity: 0.59;
  margin: 0 0 0.125rem 0;
  font-size: 0.625rem;

  color: ${({ priceAnnounced, theme }) =>
    priceAnnounced ? theme.secondary.base : theme.secondary[1]};
`

export const Big = styled(Box)`
  font-size: 0.8125rem;
  line-height: 1rem;
`

export const Pip = styled(Box)`
  font-size: 2.125rem;
  line-height: 2.5rem;
  margin: 0 0.125rem;
`

export const Tenth = styled(Box)`
  margin: 0.125rem 0;
  align-self: flex-end;
`

export const Price = styled.div<{ disabled: boolean }>`
  height: 2.1rem;
  display: flex;
  justify-content: center;
  align-items: center;

  ${({ disabled }) => (disabled ? 'opacity: 0.3' : '')}
`

export const BigWrapper = styled.div`
  height: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
`

export const ExpiredPrice = styled.div`
  color: ${({ theme }) => theme.colors.spectrum.uniqueCollections.Sell.base};
  font-size: 9px;
  text-transform: uppercase;
`
