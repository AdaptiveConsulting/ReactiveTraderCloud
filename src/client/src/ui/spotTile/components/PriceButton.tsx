import React from 'react'
import { Flex } from 'rt-components'
import { Direction } from 'rt-types'
import { styled, withDefaultProps } from 'rt-util'

const hoverColors = {
  [Direction.Buy]: '#F94C4C',
  [Direction.Sell]: '#5f94f5'
}

export const TradeButton = styled('button')<{ direction: string }>`
  background-color: ${({ theme: { background } }) => background.backgroundSecondary};
  color: ${({ theme: { text } }) => text.textPrimary};
  border-radius: 3px;
  transition: background-color 0.2s ease;
  cursor: pointer;
  border: none;
  outline: none;
  padding: 0.5rem 1.5rem;

  .spot-tile:hover & {
    background-color: ${({ theme: { background } }) => background.backgroundPrimary};
  }

  .spot-tile:hover &:hover {
    background-color: ${({ direction }) => hoverColors[direction]};
    color: ${({ theme: { text } }) => text.white};
  }
`

const Box = styled('div')`
  padding: 0;
  margin: 0;
`

const DirectionLabel = styled(Box)`
  color: ${({ theme: { text } }) => text.textMeta};
  margin: 0 0 0.125rem 0;
  font-size: 0.625rem;
`

const Big = styled(Box)`
  font-size: ${p => p.theme.fontSize.h4};
`

const Pip = styled(Box)`
  font-size: ${p => p.theme.fontSize.h2};
  margin: 0 0.125rem;
`

const Tenth = styled(Box)`
  margin: 0.125rem 0;
  align-self: flex-end;
`
const defaultProps: PriceButtonProps = {
  big: 0,
  pip: 0,
  tenth: 0,
  rawRate: 0,
  direction: Direction.Buy,
  handleClick: () => {}
}

interface PriceButtonProps {
  big: number
  pip: number
  tenth: number
  rawRate: number
  direction: Direction
  handleClick: (direction: Direction) => void
}

const renderPips = (pips: number) => (pips.toString().length === 1 ? `0${pips}` : pips)
const getBigFigureDisplay = (bigFigure: number, rawRate: number) =>
  bigFigure === Math.floor(rawRate) ? `${bigFigure}.` : bigFigure.toString()
const renderBigFigureDisplay = (bigFigureDisplay: string) =>
  bigFigureDisplay.toString().length === 3 ? `${bigFigureDisplay}0` : bigFigureDisplay

const PriceButtonComp: React.SFC<PriceButtonProps> = ({ big, pip, tenth, rawRate, direction, handleClick }) => {
  const bigFigure = getBigFigureDisplay(big, rawRate)
  return (
    <TradeButton direction={direction} onClick={() => handleClick(direction)}>
      <Flex height="34px" direction="row" justifyContent="center" alignItems="center">
        <Flex height="100%" direction="column" justifyContent="center">
          <DirectionLabel>{direction.toUpperCase()}</DirectionLabel>
          <Big>{renderBigFigureDisplay(bigFigure)}</Big>
        </Flex>
        <Pip>{renderPips(pip)}</Pip>
        <Tenth>{tenth}</Tenth>
      </Flex>
    </TradeButton>
  )
}

export default withDefaultProps(defaultProps, PriceButtonComp)
