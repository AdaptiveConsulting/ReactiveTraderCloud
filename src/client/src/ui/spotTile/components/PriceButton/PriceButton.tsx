import React from 'react'
import { Direction } from 'rt-types'
import {
  TradeButton,
  Price,
  BigWrapper,
  DirectionLabel,
  Big,
  Pip,
  Tenth,
  ExpiredPrice,
} from './styled'

interface PriceButtonProps {
  big?: number
  pip?: number
  tenth?: number
  rawRate?: number
  direction?: Direction
  handleClick?: () => void
  priceAnnounced?: boolean
  disabled?: boolean
  expired?: boolean
}

const renderPips = (pips: number) => (pips.toString().length === 1 ? `0${pips}` : pips)
const getBigFigureDisplay = (bigFigure: number, rawRate: number) =>
  bigFigure === Math.floor(rawRate) ? `${bigFigure}.` : bigFigure.toString()
const renderBigFigureDisplay = (bigFigureDisplay: string) =>
  bigFigureDisplay.toString().length === 3 ? `${bigFigureDisplay}0` : bigFigureDisplay

const PriceButtonComp: React.FC<PriceButtonProps> = ({
  big = 0,
  pip = 0,
  tenth = 0,
  rawRate = 0,
  direction = Direction.Buy,
  handleClick = () => {},
  priceAnnounced,
  disabled = false,
  expired = false,
}) => {
  const bigFigure = getBigFigureDisplay(big, rawRate)
  return (
    <TradeButton
      direction={direction}
      onClick={handleClick}
      priceAnnounced={priceAnnounced}
      disabled={disabled}
    >
      <Price disabled={disabled}>
        <BigWrapper>
          <DirectionLabel>{direction.toUpperCase()}</DirectionLabel>
          <Big>{renderBigFigureDisplay(bigFigure)}</Big>
        </BigWrapper>
        <Pip>{renderPips(pip)}</Pip>
        <Tenth>{tenth}</Tenth>
      </Price>
      {expired && <ExpiredPrice>Expired</ExpiredPrice>}
    </TradeButton>
  )
}

export default PriceButtonComp
