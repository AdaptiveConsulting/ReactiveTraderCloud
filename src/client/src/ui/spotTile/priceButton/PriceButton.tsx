import * as React from 'react'
import * as classnames from 'classnames'
import './price-button.scss'

interface PriceButtonProps {
  className: string
  direction: { name: 'Buy' | 'Sell' | string }
  rate: {
    pips: any
    bigFigure: number
    pipFraction: number
    rawRate: number,
  },
  onExecute: () => void
}

// tslint:disable-next-line:variable-name
const PriceButton = (props: PriceButtonProps) => {
  const { direction, rate } = props
  const classes = classnames('price-button', props.className)
  const paddedPips = rate.pips.toString().length === 1
    ? `0${rate.pips}`
    : rate.pips

  let bigFigureDisplay = rate.bigFigure === Math.floor(rate.rawRate)
    ? `${rate.bigFigure}.`
    : rate.bigFigure

  if (bigFigureDisplay.toString().length === 3) {
    bigFigureDisplay += '0'
  }
  return (
    <div className={classes} onClick={() => props.onExecute()}>
      <span className="price-button__wrapper">
        <span className="price-button__big-figure">
          <span className="price-button__direction">
            {direction.name}
          </span><br />
          {bigFigureDisplay}
        </span>
        <span className="price-button__pip">{paddedPips}</span>
        <span className="price-button__tenth">{rate.pipFraction}</span>
      </span>
    </div>
  )
}

export default PriceButton
