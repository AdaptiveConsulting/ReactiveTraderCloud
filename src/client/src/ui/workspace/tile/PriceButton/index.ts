export { default as PriceButton } from './PriceButton'
