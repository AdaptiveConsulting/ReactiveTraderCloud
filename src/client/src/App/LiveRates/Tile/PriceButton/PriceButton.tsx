import { getPrice$ } from "@/services/prices"
import { Direction } from "@/services/trades"
import {
  customNumberFormatter,
  significantDigitsNumberFormatter,
  DECIMAL_SEPARATOR,
} from "@/utils/formatNumber"
import { map, switchMap } from "rxjs/operators"
import { bind } from "@react-rxjs/core"
import { CenteringContainer } from "@/components/CenteringContainer"
import { AdaptiveLoader } from "@/components/AdaptiveLoader"
import { sendExecution } from "../Tile.state"
import { useTileCurrencyPair } from "../Tile.context"
import { useRfqState, QuoteStateStage } from "../Rfq"
import {
  TradeButton,
  Price,
  DirectionLabel,
  Big,
  Pip,
  Tenth,
  QuotePriceLoading,
  ExpiredPrice,
  PriceButtonDisabledPlaceholder,
  Icon,
} from "./PriceButton.styles"
import { of } from "rxjs"
import { useIsNotionalValid } from "../Notional/Notional"
import { getRfqPayload$, QuoteState, useIsRfq } from "../Rfq/Rfq.state"
import { CurrencyPair } from "@/services/currencyPairs"

const getPriceByDirection$ = (symbol: string, direction: Direction) =>
  getPrice$(symbol).pipe(
    map(({ bid, ask }) => (direction === Direction.Buy ? ask : bid)),
    map((price) => ({ isExpired: false, price })),
  )

const [usePrice, getPriceDirection$] = bind(
  (symbol: string, direction: Direction) =>
    getRfqPayload$(symbol).pipe(
      switchMap((payload) =>
        payload
          ? of({
              ...payload,
              price:
                direction === Direction.Buy
                  ? payload.rfqResponse.price.ask
                  : payload.rfqResponse.price.bid,
            })
          : getPriceByDirection$(symbol, direction),
      ),
    ),
)

export const priceButton$ = (direction: Direction) => (symbol: string) =>
  getPriceDirection$(symbol, direction)

const formatSimple = customNumberFormatter()
const formatTo2Digits = significantDigitsNumberFormatter(2)
const formatTo3Digits = significantDigitsNumberFormatter(3)
const formatToMin2IntDigits = customNumberFormatter({
  minimumIntegerDigits: 2,
})

export type PriceButtonProps = {
  direction: Direction
  price: number
  currencyPair: CurrencyPair
  priceAnnounced: boolean
  disabled: boolean
  isExpired: boolean
  isStatic?: boolean
  onClick: () => void
}

export const PriceButtonInner = ({
  direction,
  price,
  currencyPair,
  priceAnnounced,
  disabled,
  isExpired,
  isStatic,
  onClick,
}: PriceButtonProps) => {
  const { pipsPosition, ratePrecision } = currencyPair
  const rateString = price.toFixed(ratePrecision)
  const [wholeNumber, fractions_] = rateString.split(".")
  const fractions = fractions_ || "00000"

  const pip = formatToMin2IntDigits(
    Number(fractions.substring(pipsPosition - 2, pipsPosition)),
  )
  const tenth = formatSimple(
    Number(fractions.substring(pipsPosition, pipsPosition + 1)),
  )

  const bigFigureNumber = Number(
    wholeNumber + "." + fractions.substring(0, pipsPosition - 2),
  )
  let bigFigure =
    bigFigureNumber < 1 && pipsPosition === 4
      ? formatTo2Digits(bigFigureNumber)
      : formatTo3Digits(bigFigureNumber)
  if (bigFigureNumber === Math.floor(bigFigureNumber))
    bigFigure += DECIMAL_SEPARATOR

  return (
    <TradeButton
      direction={direction}
      onClick={onClick}
      priceAnnounced={priceAnnounced}
      disabled={disabled}
      isStatic={isStatic}
    >
      <Price disabled={disabled}>
        <CenteringContainer>
          <DirectionLabel>{direction.toUpperCase()}</DirectionLabel>
          <Big>{price ? bigFigure : "-"}</Big>
        </CenteringContainer>
        {price && (
          <>
            <Pip>{pip}</Pip>
            <Tenth>{tenth}</Tenth>
          </>
        )}
      </Price>
      {isExpired && (
        <ExpiredPrice data-testid="expireLabel">Expired</ExpiredPrice>
      )}
    </TradeButton>
  )
}

const PriceButtonContainer = ({
  direction,
  rfqQuoteState,
}: {
  direction: Direction
  rfqQuoteState: QuoteState
}) => {
  const currencyPair = useTileCurrencyPair()
  const { price, isExpired } = usePrice(currencyPair.symbol, direction)
  const isRfq = useIsRfq()
  const isNotionalValid = useIsNotionalValid()
  const disabled =
    price === 0 ||
    !isNotionalValid ||
    (isRfq && rfqQuoteState.stage !== QuoteStateStage.Received)

  return (
    <PriceButtonInner
      direction={direction}
      price={price}
      currencyPair={currencyPair}
      onClick={() => {
        sendExecution(currencyPair.symbol, direction)
      }}
      priceAnnounced={rfqQuoteState.stage === QuoteStateStage.Received}
      disabled={disabled}
      isExpired={isExpired}
    />
  )
}

export const AwaitingPriceButton = () => (
  <QuotePriceLoading>
    <AdaptiveLoader size={16} />
    Awaiting Price
  </QuotePriceLoading>
)

export const PriceButton = ({ direction }: { direction: Direction }) => {
  const rfqState = useRfqState()

  return rfqState.stage === QuoteStateStage.Requested ? (
    <AwaitingPriceButton />
  ) : (
    <PriceButtonContainer direction={direction} rfqQuoteState={rfqState} />
  )
}

export const PriceUnavailableButton = () => (
  <PriceButtonDisabledPlaceholder>
    <Icon className="fas fa-ban fa-flip-horizontal" />
    Pricing Unavailable
  </PriceButtonDisabledPlaceholder>
)
