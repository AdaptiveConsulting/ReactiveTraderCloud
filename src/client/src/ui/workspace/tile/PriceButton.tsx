import React from 'react'
import classnames from 'classnames'
import './priceButton.scss'

const direction = {
  Buy: 'Buy',
  Sell: 'Sell',
}

interface PriceButtonProps {
  className: string
  direction: { name: string, Buy: string, Sell: string }
  rate: {
    pips: any
    bigFigure: number
    pipFraction: number
    rawRate: number
  },
  onExecute: () => void
}

const PriceButton = (props: PriceButtonProps) => {
  const { direction, rate } = props;

  let classes = classnames(
    'price-button',
    props.className
  );
  let paddedPips = rate.pips.toString().length === 1
    ? `0${rate.pips}`
    : rate.pips;
  let bigFigureDisplay = rate.bigFigure === Math.floor(rate.rawRate)
    ? `${rate.bigFigure}.`
    : rate.bigFigure;
  if (bigFigureDisplay.toString().length === 3) {
    bigFigureDisplay += '0'
  }
  return (
    <div className={classes} onClick={() => props.onExecute()}>
      <span className="price-button__wrapper">
        <span className="price-button__big-figure">
          <span className="price-button__direction">
            {direction.name}
          </span><br />
          {bigFigureDisplay}
        </span>
        <span className="price-button__pip">{paddedPips}</span>
        <span className="price-button__tenth">{rate.pipFraction}</span>
      </span>
    </div>
  )
}

export default PriceButton
