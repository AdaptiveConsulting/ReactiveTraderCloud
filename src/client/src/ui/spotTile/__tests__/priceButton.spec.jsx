/* tslint:disable */

import * as React from 'react'
import PriceButton from '../priceButton'
import ShallowRenderer from 'react-test-renderer/shallow'
import { mount } from 'enzyme'

function getButtonProps (type, action) {
  const classNameType = type === 'Sell' ? 'bid' : 'ask'

  return {
    className: `spot-tile__price spot-tile__price--${classNameType}`,
    direction: type,
    rate: {
      pips: 5,
      bigFigure: 100,
      pipFraction: 8,
      rawRate: 123,
    },
    onExecute: action,
  }
}

describe('PriceButton', () => {
  let mockAction

  beforeEach(() => {
    mockAction = jest.fn(() => {})
  })
  
  test('renders correct markup for bid', () => {
    const shallowRenderer = new ShallowRenderer()
    const component = <PriceButton {...getButtonProps('bid', mockAction)} />
    shallowRenderer.render(component)

    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })

  test('renders correct markup for ask', () => {
    const shallowRenderer = new ShallowRenderer()
    const component = <PriceButton {...getButtonProps('ask', mockAction)} />
    shallowRenderer.render(component)

    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot()
  })

  test('onExecute action works', () => {
    const component = <PriceButton {...getButtonProps('ask', mockAction)} />
    mount(component).simulate('click')

    expect(mockAction).toBeCalled()
  })
})
