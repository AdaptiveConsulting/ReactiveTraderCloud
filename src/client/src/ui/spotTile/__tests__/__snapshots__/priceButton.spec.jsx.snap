// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceButton renders correct markup for ask 1`] = `
<div
  className="price-button spot-tile__price spot-tile__price--ask"
  onClick={[Function]}
>
  <span
    className="price-button__wrapper"
  >
    <span
      className="price-button__big-figure"
    >
      <span
        className="price-button__direction"
      >
        ask
      </span>
      <br />
      1000
    </span>
    <span
      className="price-button__pip"
    >
      05
    </span>
    <span
      className="price-button__tenth"
    >
      8
    </span>
  </span>
</div>
`;

exports[`PriceButton renders correct markup for bid 1`] = `
<div
  className="price-button spot-tile__price spot-tile__price--ask"
  onClick={[Function]}
>
  <span
    className="price-button__wrapper"
  >
    <span
      className="price-button__big-figure"
    >
      <span
        className="price-button__direction"
      >
        bid
      </span>
      <br />
      1000
    </span>
    <span
      className="price-button__pip"
    >
      05
    </span>
    <span
      className="price-button__tenth"
    >
      8
    </span>
  </span>
</div>
`;
